### ISSUE TITLE
Support Cucumber Expressions for Optional Text

### ISSUE THREAD

**Issue by loganvolkers** [alexkrechik/VSCucumberAutoComplete#212]
**Feature: Support for Cucumber Optional Text**
  As a cucumber developer 
  I want to be able to use this extension with steps that use cucumber expression optional text
  So that I get the power this extension offers in autocompleting and finding steps
  And I can use the concise syntax for optional text, and don't have to use regexes

 Cucumber expression optional text is described on their docs here: https://docs.cucumber.io/cucumber/cucumber-expressions/

-   **Scenario: Single optional character**
-   Given that I have a feature such as:
 ```
     Scenario: Delete a user
         Given a user
         When I delete that user
         Then that user should be deleted
 ```
-  And a step definition in typescript:
 ```
 When('I delete(d) that user', async function (this: World){
     this.setState({
         lastResponse: await API.deleteUser(this.state.thatUser)
     });
 })
 ```
-   When I open the feature file
-   Then I expect the step to be found
-   And I shouldn't see an error

Instead, it seems like it can't find steps that use the optional format for matching steps.

![image](https://user-images.githubusercontent.com/1157086/45332750-4b829080-b527-11e8-8acd-dbaf4af77e71.png)

![image](https://user-images.githubusercontent.com/1157086/45332754-55a48f00-b527-11e8-842b-5278b370f30f.png)

**Comment by alexkrechik** [comment]
I agree, that this issue should be supported and I'll review all the cucumber expressions and realize all the new features that are not supported yet in the nearest release. Actually, to get no such warnings until new version relized, you can use `cucumberautocomplete.customParameters` option to hack this like:
```
"cucumberautocomplete.customParameters": [
        {
            "parameter":"(d)",
            "value":"(d|)"
        },
    ],
```

**Comment by loganvolkers** [comment]
Thanks for the quick and helpful response.

Unfortunately we use a lot of optional expressions. I like the hack, but it
looks like it would be time consuming to maintain.

**Comment by alexkrechik** [comment]
Should be supported in 2.13.0

**Comment by loganvolkers** [comment]
Cool, good work Alex
